From c875a81b73007ac2b2142a00824166b27bc72f59 Mon Sep 17 00:00:00 2001
From: Shilin Victor <chrono.monochrome@gmail.com>
Date: Fri, 6 Jan 2017 14:31:57 +0700
Subject: [PATCH 24/29] Revert "ueventd: relabel block devices nodes when
 processing subsequent add events"

This reverts commit 721c9ce4a55c13716f263dc7a9cbda92b349b4b7.

Change-Id: I8c362d68f33f9cd4236c80d773d3c5f38f935523
---
 init/devices.cpp | 21 ++++++---------------
 1 file changed, 6 insertions(+), 15 deletions(-)

diff --git a/init/devices.cpp b/init/devices.cpp
index 29397b1..98cbae7 100644
--- a/init/devices.cpp
+++ b/init/devices.cpp
@@ -245,11 +245,7 @@ static void make_device(const char *path,
 
     mode = get_device_perm(path, links, &uid, &gid) | (block ? S_IFBLK : S_IFCHR);
 
-    if (selabel_lookup_best_match(sehandle, &secontext, path, links, mode)) {
-        ERROR("Device '%s' not created; cannot find SELinux label (%s)\n",
-                path, strerror(errno));
-        return;
-    }
+    selabel_lookup_best_match(sehandle, &secontext, path, links, mode);
     setfscreatecon(secontext);
 
     dev = makedev(major, minor);
@@ -259,19 +255,14 @@ static void make_device(const char *path,
      * racy. Fixing the gid race at least fixed the issue with system_server
      * opening dynamic input devices under the AID_INPUT gid. */
     setegid(gid);
-    /* If the node already exists update its SELinux label to handle cases when
-     * it was created with the wrong context during coldboot procedure. */
-    if (mknod(path, mode, dev) && (errno == EEXIST)) {
-        if (lsetfilecon(path, secontext)) {
-            ERROR("Cannot set '%s' SELinux label on '%s' device (%s)\n",
-                    secontext, path, strerror(errno));
-        }
-    }
+    mknod(path, mode, dev);
     chown(path, uid, -1);
     setegid(AID_ROOT);
 
-    freecon(secontext);
-    setfscreatecon(NULL);
+    if (secontext) {
+        freecon(secontext);
+        setfscreatecon(NULL);
+    }
 }
 
 static void add_platform_device(const char *path)
-- 
2.9.3

